File: CarRentalEntities/DO/BranchOperations/BranchOpeningHours.cs
using CarRentalEntities.IEntity;

namespace DO;

public class BranchOpeningHours : IEntity
{
    public int BranchOpeningHoursId { get; set; }
    public DayOfWeek DayOfWeek { set; get; }

    public Branch Branch { set; get; }

    public int BranchId { set; get; }

    /// <summary>
    /// 
    /// </summary>
    public TimeOnly StartTime { set; get; }

    /// <summary>
    /// 
    /// </summary>
    public TimeOnly EndTime { set; get; }

}

File: UI/CarRental/src/app/shard/models/branch/branch-opening-hours.ts
import { DayOfWeek } from '../enums/enums';

export class BranchOpeningHours {
  branchOpeningHoursId!: number;
  startDay!: DayOfWeek;
  endDay!: DayOfWeek;
  startTime!: Date;
  endTime!: Date;
}
